lib/features/contact/presentation/contact_controller.dart

```diff
@@ -0,0 +1,150 @@
+  late GetAutoCompleteInteractor _getAutoCompleteInteractor;
+
+  late Debouncer<String> _deBouncerTime;
+  late AccountId _accountId;
+  late Session _session;
+
+  EmailAddress? contactSelected;
+
+  @override
+  void onInit() {
+    super.onInit();
+    _initializeDebounceTimeTextSearchChange();
+  }
+
+  @override
+  void onReady() async {
+    textInputSearchFocus.requestFocus();
+    final arguments = Get.arguments;
+    if (arguments is ContactArguments) {
+      _accountId = arguments.accountId;
+      _session = arguments.session;
+      final listContactSelected = arguments.listContactSelected;
+      if (listContactSelected.isNotEmpty) {
+        contactSelected = EmailAddress(listContactSelected.first, listContactSelected.first);
+      }
+    }
+    if (!BuildUtils.isWeb) {
+      Future.delayed(
+        const Duration(milliseconds: 500),
+        () => _checkContactPermission());
+    }
+    super.onReady();
+  }
+
+  @override
+  void onClose() {
+    textInputSearchController.dispose();
+    _deBouncerTime.cancel();
+    super.onClose();
+  }
+
+  void _initializeDebounceTimeTextSearchChange() {
+    _deBouncerTime = Debouncer<String>(
+        const Duration(milliseconds: 500),
+        initialValue: '');
+    _deBouncerTime.values.listen((value) async {
+      searchQuery.value = SearchQuery(value);
+      _searchContactByNameOrEmail(searchQuery.value.value);
+    });
+  }
+
+  void onTextSearchChange(String text) {
+    _deBouncerTime.value = text;
+  }
+
+  void clearAllTextInputSearchForm() {
+    textInputSearchController.clear();
+    searchQuery.value = SearchQuery.initial();
+    textInputSearchFocus.requestFocus();
+  }
+
+  void _checkContactPermission() async {
+    final permissionStatus = await Permission.contacts.status;
+    if (permissionStatus.isGranted) {
+      _contactSuggestionSource = ContactSuggestionSource.all;
+    } else if (!permissionStatus.isPermanentlyDenied) {
+      final requestedPermission = await Permission.contacts.request();
+      _contactSuggestionSource = requestedPermission == PermissionStatus.granted
+          ? ContactSuggestionSource.all
+          : _contactSuggestionSource;
+    }
+  }
+
+  void _searchContactByNameOrEmail(String query) async {
+    log('ContactController::_searchContactByNameOrEmail(): query: $query');
+    final listContact = await _getAutoCompleteSuggestion(query);
+    listContactSearched.value = listContact;
+  }
+
+  Future<List<EmailAddress>> _getAutoCompleteSuggestion(String query) async {
+    if(_session.hasSupportTmailAutoComplete != true) return <EmailAddress>[];
+
+    if (!Get.isRegistered<GetAutoCompleteWithDeviceContactInteractor>() ||
+        !Get.isRegistered<GetAutoCompleteInteractor>()) {
+      AutoCompleteBindings().dependencies();
+    }
+
+    _getAutoCompleteWithDeviceContactInteractor = Get.find<GetAutoCompleteWithDeviceContactInteractor>();
+    _getAutoCompleteInteractor = Get.find<GetAutoCompleteInteractor>();
```

Comment:
please take care the case: `Get can not found the interactor`

Reply:
There is a check in the previous statement. Also if it is not already injected, it will be injected again. So it always gets the interactor.

Reply:
> Also if it is not already injected, it will be injected again

really?

Reply:
<img width="853" alt="Screen Shot 2022-09-20 at 14 51 40" src="https://user-images.githubusercontent.com/80730648/191199592-78d678b5-0c1f-4cfe-89c3-230e5a3946ff.png">